superchain/params.go

```diff
@@ -71,3 +71,75 @@ var OPSepoliaDev0L2OOParams = L2OOParams{
 	L2BlockTime:               big.NewInt(2),
 	FinalizationPeriodSeconds: big.NewInt(12),
 }
+
+type BigIntAndBounds struct {
+	Value  *big.Int
+	Bounds [2]*big.Int
+}
+
+type Uint32AndBounds struct {
+	Value  uint32
+	Bounds [2]uint32
+}
+
+type PreEcotoneGasPriceOracleParams struct {
+	Decimals *big.Int
+	Overhead *big.Int
+	Scalar   *big.Int
+}
+
+type EcotoneGasPriceOracleParams struct {
+	Decimals          *big.Int
+	BlobBaseFeeScalar uint32
+	BaseFeeScalar     uint32
+}
+
+type PreEcotoneGasPriceOracleParamsWithBounds struct {
+	Decimals BigIntAndBounds
+	Overhead BigIntAndBounds
+	Scalar   BigIntAndBounds
+}
+
+type EcotoneGasPriceOracleParamsWithBounds struct {
+	Decimals          BigIntAndBounds
+	BlobBaseFeeScalar Uint32AndBounds
+	BaseFeeScalar     Uint32AndBounds
+}
+
+type UpgradeFilter struct {
+	PreEcotone PreEcotoneGasPriceOracleParamsWithBounds
+	Ecotone    EcotoneGasPriceOracleParamsWithBounds
+}
+
+func makeBigIntAndBounds(value int64, bounds [2]int64) BigIntAndBounds {
+	return BigIntAndBounds{big.NewInt(value), [2]*big.Int{big.NewInt(bounds[0]), big.NewInt(bounds[1])}}
+}
+
+var GasPriceOracleParams = map[string]UpgradeFilter{
+	"mainnet": {
+		PreEcotone: PreEcotoneGasPriceOracleParamsWithBounds{
+			Decimals: makeBigIntAndBounds(6, [2]int64{6, 6}),
+			Overhead: makeBigIntAndBounds(188, [2]int64{188, 188}),
+			Scalar:   makeBigIntAndBounds(684_000, [2]int64{684_000, 684_000}),
+		},
+	},
+	"sepolia": {
+		PreEcotone: PreEcotoneGasPriceOracleParamsWithBounds{
+			Decimals: makeBigIntAndBounds(6, [2]int64{6, 6}),
+			Overhead: makeBigIntAndBounds(188, [2]int64{188, 2_100}),
+			Scalar:   makeBigIntAndBounds(684_000, [2]int64{684_000, 1_000_000}),
+		},
+	},
+	"goerli": {
+		PreEcotone: PreEcotoneGasPriceOracleParamsWithBounds{
+			Decimals: makeBigIntAndBounds(6, [2]int64{6, 6}),
+			Overhead: makeBigIntAndBounds(2_100, [2]int64{2_100, 2_100}),
+			Scalar:   makeBigIntAndBounds(100_000, [2]int64{100_000, 100_000}),
+		},
+		Ecotone: EcotoneGasPriceOracleParamsWithBounds{
+			Decimals:          makeBigIntAndBounds(6, [2]int64{6, 6}),
+			BlobBaseFeeScalar: Uint32AndBounds{862_000, [2]uint32{862_000, 862_000}},
+			BaseFeeScalar:     Uint32AndBounds{7600, [2]uint32{7600, 7600}},
```

Comment:
I went ahead and supported `PreEcotone` and `Ecotone` networks for now. At the reviewer's request I'm happy to remove the `PreEcotone` support to simplify / reduce boilerplate. This data structure could be a useful scaffolding for supporting future upgrades, though, so there's an argument for keeping it in.